ingress/scrape_citation_contexts.py

```diff
@@ -0,0 +1,152 @@
+
+
+class OneTimeTokenizer(Tokenizer):
+    """
+    Wrap the CourtListener tokenizer to save tokenization results.
+    """
+    base_tokenizer: Tokenizer
+
+    def __init__(self, base_tokenizer: Tokenizer):
+        self.base_tokenizer = base_tokenizer
+        self.words = []
+        self.cit_toks = []
+
+    def tokenize(self, text: str):
+        if not self.words or self.cit_toks:
+            # some of the static methods in AhocorasickTokenizer don't like children.
+            self.words, self.cit_toks = self.base_tokenizer.tokenize(text)
+        return self.words, self.cit_toks
+
+
+class CitationContextScraper:
+    eyecite_tokenizer: Tokenizer
+    reporter_resource_dict: dict
+    process_pool_size: int
+
+    def __init__(self, process_pool_size=1):
+        self.process_pool_size = process_pool_size
+
+    def scrape_contexts(self):
+        Logger.info("Constructing reporter to resource_id dict...")
+        with get_session() as s:
+            self.reporter_resource_dict = self.__get_reporter_resource_dict(s)
+        try:
+            Logger.info("Initializing Hyperscan tokenizer...")
+            # noinspection PyPackageRequirements,PyUnresolvedReferences
+            import hyperscan
+            self.eyecite_tokenizer = HyperscanTokenizer(cache_dir=HYPERSCAN_TMP_PATH)
+        except:
+            Logger.info("Failed to initialize hyperscan, using Ahocorasick...")
+            self.eyecite_tokenizer = AhocorasickTokenizer()
+        with Pool(self.process_pool_size) as p:
+            p.map(self.populate_jurisdiction_db_context, JURISDICTIONS)
+
+    def __populate_db_contexts_for_opinion(self, session: Session, opinion: Opinion, reporter_resource_dict: dict,
```

Comment:
why are double underscores/name mangling required here?

Reply:
Just to mark this method as private

Reply:
I understand single underscore to be sufficient for that purpose, double underscore triggers some name-mangling shit we probably shouldn't resort to unless specifically necessary.